UDV-Core/src/Extensions/Contribute/src/ContributeController.js

```diff
@@ -0,0 +1,225 @@
+   * Gets the current visualization data set by the user and
+   *  moves camera to this position
+   */
+  //=============================================================================
+  this.moveDoc = function moveDoc(){
+    this.chosenPosition.x = document.getElementById("setPosX").value;
+    this.chosenPosition.y = document.getElementById("setPosY").value;
+    this.chosenPosition.z = document.getElementById("setPosZ").value;
+    this.chosenQuaternion.x = document.getElementById("quatX").value;
+    this.chosenQuaternion.y = document.getElementById("quatY").value;
+    this.chosenQuaternion.z = document.getElementById("quatZ").value;
+    this.chosenQuaternion.w = document.getElementById("quatW").value;
+    this.documentController.controls.initiateTravel(this.chosenPosition,"auto",
+                                                    this.chosenQuaternion,true);
+  }
+
+  /**
+  * Colors the form fields if value is not set
+  */
+  //=============================================================================
+  this.colorField = function colorField(id, color){
+    if(color == true){
+    document.getElementById(id).style.border = "1px solid red";
+  }
+  else {
+    document.getElementById(id).style.border = "1px solid white";
+    }
+  }
+
+  /**
+   * Checks if visualization data is provided
+   */
+  //=============================================================================
+  this.visuDataVerification = function visuDataVerification(){
+    if(this.docPos == null || this.docQuat == null){
+      this.validPosition = false;
+      alert('Please chose document position and save');
+      this.documentCreate.showDocPositioner(false);
+    }
+    else {
+      this.validPosition = true;
+    }
+    return this.validPosition;
+  }
+
+
+  /**
+   * Verifies form data
+   */
+  //=============================================================================
+  this.formDataVerification = function formDataVerification(){
+    var dataIsValid = true;
+    this.formData = new FormData(document.getElementById('creationForm'));
+    this.newDocData = this.formData;
+    var txt ="Please provide "; //error message if data unvalid
+
+    for (var pair of this.formData.entries() ){
+      if( pair[1] == ""){
+        if(pair[0] == "link"){
+        txt += "file ";
+        dataIsValid = false;
+      }
+      else {
+        txt += pair[0] + ", ";
+        var id = "create_"+pair[0];
+        this.colorField(id, true);
+        dataIsValid = false;
+      }
+    }
+  }
+  if(!dataIsValid){
+    alert(txt);
+  }
+  return dataIsValid;
+}
+
+/**
+  Post document if data is OK, reset all data to prepare next document creation
+ */
+//=============================================================================
+  this.documentCreation = function documentCreation(){
+
+    if (this.formDataVerification() ==true & this.visuDataVerification() == true){
+      //add visualizationdata
+      this.newDocData.append("positionX", this.docPos.x);
```

Comment:
this.newDocData is a FormData object.
It holds the data form the form and the document placement values.
Could you be more precise?

Reply:
I meant instead of manually managing the position (x,y,z) and quaternion (x,y,z) coordinates, could you use a FormData Object with 6 fields to manage them like what you do with the metadata of the document (i.e. with this.formData attribute) ?

Reply:
ok